ddl/column.go

```diff
@@ -644,10 +644,49 @@ func onSetDefaultValue(t *meta.Meta, job *model.Job) (ver int64, _ error) {
 func needChangeColumnData(oldCol, newCol *model.ColumnInfo) bool {
 	toUnsigned := mysql.HasUnsignedFlag(newCol.Flag)
 	originUnsigned := mysql.HasUnsignedFlag(oldCol.Flag)
+	switch oldCol.Tp {
+	case mysql.TypeEnum:
+		switch newCol.Tp {
```

Comment:
I wavered between `if` and `switch`. At first I use `if` then I changed it to `switch`, because I think there will be many type's comparsion. May be the `switch` is a better choose? If you want to change it. I will follow you.

Reply:
We better extract the logic to `judgeFunc(...)`. 
Keeping it the switch is ok.